qutip/core/data/make.py

```diff
@@ -119,7 +119,11 @@ def one_element_dia(shape, position, value=1.0):
     data = np.zeros((1, shape[1]), dtype=complex)
     data[0, position[1]] = value
     offsets = np.array([position[1]-position[0]])
-    return Dia((data, offsets), copy=False, shape=shape)
+    if np.lib.NumpyVersion(np.__version__) >= '2.0.0b1':
+        copy_if_needed = None
+    else:
+        copy_if_needed = False
+    return Dia((data, offsets), copy=copy_if_needed, shape=shape)
```

Comment:
Our interface shouldn't depend on the underlying version of numpy, so perhaps this logic needs to live in `Dia.__init__`?

Reply:
I would prefer to just let the back-ends manage it.
Only this call need fix for our tests, but we use `np.array` close to a hundred times and we have to think about jax, cupy, etc. too. I don't think we should fight to make them all works the same...

Reply:
By "our interface" I meant just "Dia.__init__" not the dataclass interface as a whole. If we don't change this inside `__init__` then our users also need to do this check if they call `Dia(...)`.

Reply:
From discussion with @Ericgig: Perhaps we can do something like:
```
if numpy < 2:
    copy = bool(copy)
```
in the `__init__` for CSR, Dense and Dia.

Reply:
Thanks for implementing, @Ericgig.